pkg/auth/providers/azure/azure_provider_test.go

```diff
@@ -0,0 +1,192 @@
+
+			schemas := types.NewSchemas()
+			schemas.AddSchemas(managementschema.AuthSchemas)
+
+			rw := &writer.EncodingResponseWriter{
+				ContentType: "application/json",
+				Encoder:     types.JSONEncoder,
+			}
+			rr := httptest.NewRecorder()
+			r := &types.APIContext{
+				Schemas:        schemas,
+				Request:        req,
+				Response:       rr,
+				ResponseWriter: rw,
+				Version:        &managementschema.Version,
+			}
+
+			provider := azureProvider{}
+			err = provider.ConfigureTest("configureTest", nil, r)
+			assert.NoError(t, err)
+
+			res := rr.Result()
+			defer res.Body.Close()
+
+			var output v3.AzureADConfigTestOutput
+			err = json.NewDecoder(res.Body).Decode(&output)
+			assert.NoError(t, err)
+			assert.Equal(t, test.expectedRedirectURL, output.RedirectURL)
+		})
+	}
+
+}
+
+// TestTransformToAuthProvider inspects the Redirect URL during login.
+func TestTransformToAuthProvider(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name                string
+		authConfig          map[string]interface{}
+		expectedRedirectUrl string
+	}{
+		{
+			name: "redirect URL for Microsoft Graph",
+			authConfig: map[string]interface{}{
+				"enabled":    true,
+				"accessMode": "unrestricted",
+				"metadata": map[string]interface{}{
+					"annotations": map[string]interface{}{
+						"auth.cattle.io/azuread-endpoint-migrated": "true",
+					},
+				},
+				"endpoint":          "https://login.microsoftonline.com/",
+				"graphEndpoint":     "https://graph.microsoft.com",
+				"tokenEndpoint":     "https://login.microsoftonline.com/tenant123/oauth2/v2.0/token",
+				"authEndpoint":      "https://login.microsoftonline.com/tenant123/oauth2/v2.0/authorize",
+				"tenantId":          "tenant123",
+				"applicationId":     "app123",
+				"applicationSecret": "secret123",
+				"rancherUrl":        "https://myrancher.com",
+			},
+			expectedRedirectUrl: "https://login.microsoftonline.com/tenant123/oauth2/v2.0/authorize?client_id=app123&redirect_uri=https://myrancher.com&response_type=code&scope=openid",
+		},
+		{
+			name: "redirect URL for Azure AD Graph",
+			authConfig: map[string]interface{}{
+				"enabled":           true,
+				"accessMode":        "unrestricted",
+				"endpoint":          "https://login.microsoftonline.com/",
+				"graphEndpoint":     "https://graph.windows.net/",
+				"tokenEndpoint":     "https://login.microsoftonline.com/tenant123/oauth2/token",
+				"authEndpoint":      "https://login.microsoftonline.com/tenant123/oauth2/authorize",
+				"tenantId":          "tenant123",
+				"applicationId":     "app123",
+				"applicationSecret": "secret123",
+				"rancherUrl":        "https://myrancher.com",
+			},
+			expectedRedirectUrl: "https://login.microsoftonline.com/tenant123/oauth2/authorize?client_id=app123&redirect_uri=https://myrancher.com&resource=https://graph.windows.net/&scope=openid",
+		},
+	}
+
+	var provider azureProvider
+	for _, test := range tests {
```

Comment:
Nit: There doesn't appear to be tests for the disabled auth config (old or new) as far as the "Get from K8s" case goes.

Reply:
I am not sure what the use case would be for fetching a disabled auth config on login.

Reply:
If I'm understanding correctly, that's not what this function is testing. This is testing the `TransformToProvider` method. Rancher might use that only on login, but it's an exported method. So you can't rely on that only being called on login.